src/components/Form/Textarea.tsx

```diff
@@ -0,0 +1,139 @@
+  onBlur?: (event: FocusEvent<HTMLTextAreaElement>) => void;
+  /** Label */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: 'hidden' | 'visible';
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Text content for textarea */
+  children?: string;
+  /** Hint text to be shown below the component */
+  hintText?: string;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: 'default' | 'error';
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** Resize mode of the textarea
+      'both' | 'vertical' | 'horizontal' | 'none'
+      @default 'vertical' 
+   */
+  resize?: 'both' | 'vertical' | 'horizontal' | 'none';
+  optionalText?: string;
+}
+
+class BaseTextarea extends Component<TextareaProps> {
+  render() {
+    const {
+      id: propId,
+      className,
+      disabled = false,
+      children,
+      onClick,
+      labelMode,
+      labelText,
+      hintText,
+      status,
+      statusText,
+      visualPlaceholder,
+      resize: dismissResize,
+      optionalText,
+      ...passProps
+    } = this.props;
+
+    const onClickProps = !!disabled ? {} : { onMouseDown: onClick };
+
+    const hideLabel = labelMode === 'hidden';
+    const id = idGenerator(propId);
+    const statusTextId = `${idGenerator(propId)}-statusText`;
+    const hintTextId = `${idGenerator(propId)}-hintText`;
+
+    return (
+      <HtmlLabel className={classnames(baseClassName, className, {})}>
+        {hideLabel ? (
+          <VisuallyHidden>{labelText}</VisuallyHidden>
+        ) : (
+          <Paragraph>
+            <HtmlSpan className={textareaClassNames.label}>
+              {labelText}
+            </HtmlSpan>
+            {optionalText && (
+              <HtmlSpan className={textareaClassNames.optionalText}>
+                {`(${optionalText})`}
+              </HtmlSpan>
+            )}
+          </Paragraph>
+        )}
+        {hintText && (
+          <HtmlSpan className={textareaClassNames.hintText} id={hintTextId}>
+            {hintText}
+          </HtmlSpan>
+        )}
+        <HtmlTextarea
+          id={id}
+          className={textareaClassNames.textarea}
+          disabled={disabled}
+          defaultValue={children}
+          placeholder={visualPlaceholder}
+          {...passProps}
+          {...onClickProps}
+        />
+        {statusText && !disabled && (
+          <HtmlSpan className={textareaClassNames.statusText} id={statusTextId}>
+            {statusText}
+          </HtmlSpan>
+        )}
+      </HtmlLabel>
+    );
+  }
+}
+
+const StyledBaseTextarea = styled((props: TextareaProps) => (
+  <BaseTextarea {...props} />
+))`
+  & .${textareaClassNames.textarea} {
+    resize: ${(props) => props.resize || 'vertical'};
+  }
```

Comment:
This was done like this because the `resize` need to be added with CSS to the textarea.

Reply:
If we would like to add also `minWidth` and `minHeight` props for the **Textarea** component, this is the place to add those also.

Reply:
If I remember correctly, some cases have been handled by using props when generating the baseStyles, so this could be also done in baseStyles.tsx. Not sure which is better though.

Reply:
Yes, that is true. I guess it is doable to make the same logic in the baseStyles.

I picked one place where prop is looked in baseStyles for example:
https://github.com/vrk-kpa/suomifi-ui-components/blob/716065208128775fad8928ae5d0ef3bb5b196903/src/core/Text/Text.baseStyles.ts#L7-L11

Reply:
In my opinion the place to add the classnames logic does not matter that much, but I'd like to keep the css in the baseStyles file.

Reply:
The `resize` prop is now taken into account on the **baseStyles** instead in the recent commit.

Reply:
I agree that it would be a clearer and cleaner practice to keep the css in one place, even if it requires some extra lines.

Reply:
I think this one can be resolved.